app/src/main/java/com/example/mplayer/data/source/local/SongLocalDataSource.java

```diff
@@ -0,0 +1,25 @@
+package com.example.mplayer.data.source.local;
+
+import android.content.Context;
+import com.example.mplayer.data.source.UserDataSource;
+import com.example.mplayer.asynctask.GetSongAsyncTask;
+
+public class SongLocalDataSource implements UserDataSource.Local {
+    private static SongLocalDataSource sInstance;
+
+    private SongLocalDataSource() {
+
+    }
+
+    public static SongLocalDataSource getInstance() {
+        if (sInstance == null) {
+            sInstance = new SongLocalDataSource();
+        }
+        return sInstance;
+    }
+
+    @Override
+    public void getData(Context context, UserDataSource.DataCallback dataCallback) {
```

Comment:
em nên inject context vào trong LocalDataSource để sử dụng
ko nên truyền vào method ntn

Reply:
@manhpv01 anh Toản nói làm như này là sao b ơi